docker/dolfinx-complex-mode

```diff
@@ -1,6 +1,6 @@
 #!/bin/bash
-export LD_LIBRARY_PATH=/usr/local/dolfinx-complex/lib:$LD_LIBRARY_PATH
 export PATH=/usr/local/dolfinx-complex/bin:$PATH
 export PKG_CONFIG_PATH=/usr/local/dolfinx-complex/lib/pkgconfig:$PKG_CONFIG_PATH
 export PETSC_ARCH=linux-gnu-complex-32
 export PYTHONPATH=/usr/local/dolfinx-complex/lib/python3.8/dist-packages:$PYTHONPATH
+export LD_LIBRARY_PATH=/usr/local/petsc/$PETSC_ARCH/lib:/usr/local/dolfinx-complex/lib:$LD_LIBRARY_PATH
```

Comment:
Is this needed? Is the issue runtime linking issues? I would have expected CMake to take care of linking.